mods/acrylic-effect-radius-changer.wh.cpp

```diff
@@ -0,0 +1,137 @@
+*Custom blur radius (24x2)*
+
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- radius:
+  - width: 30
+  - height: 30
+  $name: Blur radius
+  $description: Set the radius of the Acrylic blur effect to any number
+*/
+// ==/WindhawkModSettings==
+
+#include <d2d1.h>
+#include <d2d1effects.h>
+#include <dcommon.h>
+
+struct {
+    int width;
+    int height;
+} settings;
+
+void LoadSettings() {
+    settings.width = Wh_GetIntSetting(L"radius.width");
+    settings.height = Wh_GetIntSetting(L"radius.height");
+}
+
+long (*CCustomBlur_BuildEffect_orig)(void* pThis, ID2D1Image* image, D2D_RECT_F*, D2D_SIZE_F*, DWORD, D2D_VECTOR_2F*, D2D_VECTOR_2F*);
+
+long CCustomBlur_BuildEffect_Hook(void* pThis, ID2D1Image* image, D2D_RECT_F* blurrect, D2D_SIZE_F* blurradius, DWORD optimization, D2D_VECTOR_2F* vector1, D2D_VECTOR_2F* vector2) {
+    blurradius->width = (float)settings.width;
+    blurradius->height = (float)settings.height;
+    Wh_Log(L"%i", settings.width);
+    return CCustomBlur_BuildEffect_orig(pThis, image, blurrect, blurradius, optimization, vector1, vector2);
+}
+
+float (*CCustomBlur_DetermineOutputScale_orig)(float, float, DWORD);
+
+float CCustomBlur_DetermineOutputScale_Hook(float f1, float f2, DWORD optimization) {
+    f1 = 0; // Makes the effect slightly less optimized, but makes it look significantly better in some cases
+    f2 = 0; // In a future update, I might try to reimplement the lost optimization.
+
+    return CCustomBlur_DetermineOutputScale_orig(f1, f2, optimization);
+}
+
+// The mod is being initialized, load settings, hook functions, and do other
+// initialization stuff if required.
+BOOL Wh_ModInit() {	
+    HMODULE dwmcore = GetModuleHandle(L"dwmcore.dll");
+    if (!dwmcore) return FALSE;
+
+    WH_FIND_SYMBOL findSymbol;
+    HANDLE findSymbolHandle = Wh_FindFirstSymbol(dwmcore, nullptr, &findSymbol);
+    if (!findSymbolHandle) {
+        Wh_Log(L"Wh_FindFirstSymbol failed");
+        return FALSE;
+    }
+    
+    void* DetermineOutputScaleAddr = nullptr;
+    void* BuildEffectAddr = nullptr;
+
+    do {
+        if (_wcsicmp(findSymbol.symbol, L"public: static float __cdecl CCustomBlur::DetermineOutputScale(float,float,enum D2D1_GAUSSIANBLUR_OPTIMIZATION)") == 0) {
+            DetermineOutputScaleAddr = findSymbol.address;
+            Wh_Log(L"symbol: %s, Addr: %i", findSymbol.symbol, findSymbol.address);
+        }
+        else if (_wcsicmp(findSymbol.symbol, L"public: long __cdecl CCustomBlur::BuildEffect(struct ID2D1Image *,struct D2D_RECT_F const &,struct D2D_SIZE_F const &,enum D2D1_GAUSSIANBLUR_OPTIMIZATION,struct D2D_VECTOR_2F const &,struct D2D_VECTOR_2F *)") == 0) {
+            BuildEffectAddr = findSymbol.address;
+            Wh_Log(L"symbol: %s, Addr: %i", findSymbol.symbol, findSymbol.address);
+        }
+        /*else {
+            Wh_Log(L"symbol: %s", findSymbol.symbol);
+        }*/
+    } while (Wh_FindNextSymbol(findSymbolHandle, &findSymbol));
```

Comment:
I suggest using `HookSymbols` of `windhawk_utils.h` for symbol resolution. It caches symbols to avoid loading and enumerating them every time, which can be slow. It's not documented properly yet but I plan on doing it soon, after releasing a new Windhawk version.

Usage example:
https://github.com/ramensoftware/windhawk-mods/blob/59d4fcd662a854e70f166ea1756cf92ca940b02e/mods/msg-box-font-fix.wh.cpp#L77-L91